EventManagement/EventAPI/Controllers/EventController.cs

```diff
@@ -24,287 +24,51 @@ public EventController(EventCatalogContext context, IConfiguration config)
-            };
-            return Ok(model);
-        }
-
-        [HttpGet("[Action]")]
-        public async Task<IActionResult> EventByCategory(
-            [FromQuery] DateTime? ValidDate,
-            [FromQuery] int CategoryId,
-            [FromQuery] int pageIndex = 0,
-            [FromQuery] int pageSize = 6)
-        {
-            var query = (IQueryable<Event>)_context.EventCatalog;
-            query = query.Where(e => e.IsCancelled == false);
-
-            if (ValidDate.HasValue)
-            {
-                query = query.Where(e => e.StartDate >= ValidDate);
-            }
-            query = query.Where(e => (_context.SubCategories.Any(s => s.SubCategoryId == e.SubCategoryId && s.CategoryId == CategoryId)));
-            
-            var eventsCount = await query.LongCountAsync();
-            var events = await query
-                                    .OrderBy(e => e.Id)
-                                    .Skip(pageSize * pageIndex)
-                                    .Take(pageSize)
-                                    .ToListAsync();
-            events = ChangePictureUrl(events);
-            var model = new PaginatedEventsViewModel
-            {
-                Data = events,
-                PageIndex = pageIndex,
-                PageSize = events.Count,
-                Count = eventsCount
-            };
-            return Ok(model);
-        }
-
-        [HttpGet("[Action]")]
-        public async Task<IActionResult> EventByFormat(
-            [FromQuery] DateTime? ValidDate,
-            [FromQuery] int FormatId,
-            [FromQuery] int pageIndex = 0,
-            [FromQuery] int pageSize = 6)
-        {
-            var query = (IQueryable<Event>)_context.EventCatalog;
-            query = query.Where(e => e.IsCancelled == false);
-
-            if (ValidDate.HasValue)
-            {
-                query = query.Where(e => e.StartDate >= ValidDate);
-            }
-
-            query = query.Where(e => e.FormatId == FormatId);
-            
-            var eventsCount = await query.LongCountAsync();
-            var events = await query
-                                    .OrderBy(e => e.Id)
-                                    .Skip(pageSize * pageIndex)
-                                    .Take(pageSize)
-                                    .ToListAsync();
-            events = ChangePictureUrl(events);
-            var model = new PaginatedEventsViewModel
-            {
-                Data = events,
-                PageIndex = pageIndex,
-                PageSize = events.Count,
-                Count = eventsCount
-            };
-
-            return Ok(model);
-        }
+        //API to retrieve Active Events
 
         [HttpGet("[Action]")]
         public async Task<IActionResult> Events(
-            [FromQuery] DateTime? ValidDate,
             [FromQuery] DateTime? startDate, 
             [FromQuery] DateTime? endDate,
             [FromQuery] string city,
```

Comment:
I'm not sure about these two lines -
            [FromQuery] string city,
            [FromQuery] string state,
            
            Does it need the "?" in case the parameter is missing in the request?
            [FromQuery] string? city,
            [FromQuery] string? state,

Reply:
[FromQuery] will give us the flexibility to run the same API even with some missing parameters. 
The nullable data types (eg: int?) are used to allow users to enter null values for a particular datatype. Since, "string" datatype allows "null" by default, we don't need to put a "?" for this data type.

Reply:
It seems reasonable at the same time if @AndrewPBartson  are agree , I go for merging pull request then we can test it .